File: algorithms/M/Maximum of Absolute Value Expression/Maximum of Absolute Value Expression.py
class Solution(object):
    def maxAbsValExpr(self, arr1, arr2):
        """
        :type arr1: List[int]
        :type arr2: List[int]
        :rtype: int
        """
        max_ppp,max_ppm,max_pmp,max_pmm=float('-inf'),float('-inf'),float('-inf'),float('-inf')
        min_ppp,min_ppm,min_pmp,min_pmm=float('inf'),float('inf'),float('inf'),float('inf')
        for i,(a,b) in enumerate(zip(arr1,arr2)):
            ppp=a+b+i
            if ppp>max_ppp:max_ppp=ppp
            if ppp<min_ppp:min_ppp=ppp
            ppm=a+b-i
            if ppm>max_ppm:max_ppm=ppm
            if ppm<min_ppm:min_ppm=ppm
            pmp=a-b+i
            if pmp>max_pmp:max_pmp=pmp
            if pmp<min_pmp:min_pmp=pmp
            pmm=a-b-i
            if pmm>max_pmm:max_pmm=pmm
            if pmm<min_pmm:min_pmm=pmm
        return max(max_ppp-min_ppp,max_ppm-min_ppm,max_pmp-min_pmp,max_pmm-min_pmm)   

File: scripts/algorithms/M/Maximum of Absolute Value Expression/Maximum of Absolute Value Expression.cpp
// Runtime: 125 ms (Top 8.84%) | Memory: 29.3 MB (Top 14.63%)
class Solution {
public:
    int maxAbsValExpr(vector<int>& arr1, vector<int>& arr2) {
        int res=0;
        int n = arr1.size();
        vector<int>v1;
        vector<int>v2;
        vector<int>v3;
        vector<int>v4;
        for(int i=0;i<n;i++)
        {
            v1.push_back(i+arr1[i]+arr2[i]);
            v2.push_back(i+arr1[i]-arr2[i]);
            v3.push_back(i-arr1[i]+arr2[i]);
            v4.push_back(i-arr1[i]-arr2[i]);
        }
        res = max(res,*max_element(v1.begin(),v1.end())-*min_element(v1.begin(),v1.end()));
        res = max(res,*max_element(v2.begin(),v2.end())-*min_element(v2.begin(),v2.end()));
        res = max(res,*max_element(v3.begin(),v3.end())-*min_element(v3.begin(),v3.end()));
        res = max(res,*max_element(v4.begin(),v4.end())-*min_element(v4.begin(),v4.end()));
        return res;
    }
};
